### PR TITLE
Add complex property path test

### PR DESCRIPTION
**Pull request by patofp**
# Description

This PR adds integration tests of complex property paths, by combining some of the defined property paths.
It also fixes some minor bugs:
- Property path DAG rewrite when some property paths.
- Return duplicates triples on some corner cases.

## Type of change

Make sure you add the correct label for the PR:

- `enhacement` if this PR adds a new feature
- `bug` if it fixes a bug
- `documentation` if it adds docs
- `breaking-change` if this PR introduces a breaking change
- `dependency-updates` if it updates dependencies

### CODE REVIEW

File: modules/engine/src/main/scala/com/gsk/kg/engine/functions/PathFrame.scala [hunk 1]
```diff
@@ -240,7 +244,7 @@ object PathFrame {
   ): Monoid[PathFrame] =
     new Monoid[PathFrame] {
       def combine(x: PathFrame, y: PathFrame): PathFrame =
-        merge(x, y).distinct
```
Comment (pepegar) [inline modules/engine/src/main/scala/com/gsk/kg/engine/functions/PathFrame.scala:243]: 👌🏻

File: modules/engine/src/main/scala/com/gsk/kg/engine/optimizer/PropertyPathRewrite.scala [hunk 2]
```diff
@@ -390,9 +465,103 @@ object PropertyPathRewrite {
...
+//      case OneOrMore(Alternative(pel, per)) =>
+//        Union(
+//          Coattr.pure(OneOrMore(pel)),
+//          Coattr.pure(OneOrMore(per))
+//        )
+//      case OneOrMore(SeqExpression(pel, per)) =>
+//        Union(
+//          Coattr.pure(OneOrMore(pel)),
+//          Coattr.pure(OneOrMore(per))
+//        )
```
Comment (pepegar) [inline modules/engine/src/main/scala/com/gsk/kg/engine/optimizer/PropertyPathRewrite.scala:523]: Do we want to leave this code commented?

**Review by pepegar — APPROVED** [review]
LGTM

File: modules/engine/src/main/scala/com/gsk/kg/engine/optimizer/PropertyPathRewrite.scala (hunk 2 above)
Comment (patofp) [reply modules/engine/src/main/scala/com/gsk/kg/engine/optimizer/PropertyPathRewrite.scala:523]: this code can be deleted 😉
